pkg/daemon/ceph/client/upgrade.go

```diff
@@ -401,6 +402,15 @@ func osdDoNothing(context *clusterd.Context, clusterInfo *ClusterInfo) bool {
 	// aio means all in one
 	aio, err := allOSDsSameHost(context, clusterInfo)
 	if err != nil {
+		// We return true so that we can continue without a retry and subsequently not test if the osd can be stopped
+		// This handles the scenario where the OSDs have been created but not yet started due to a wrong CR configuration
+		// For instance, when OSDs are encrypted and Vault is used to store encryption keys, if the KV version is incorrect during the cluster initialization
+		// the OSDs will fail to start and stay in CLBO until the CR is updated again with the correct KV version so that it can start
+		// For this scenario we don't need to go through the path where the check whether the OSD can be stopped or not, so it will always fail and make us wait for nothing
```

Comment:
Nit/discussion: for slightly better readability in the code, maybe these lines could be a little shorter?

FWIW, I use https://marketplace.visualstudio.com/items?itemName=stkb.rewrap at 100 lines.